Dataframe styling adjustments

akshayka opened the pull request.

mscolnick commented:
The groupby and aggregate can look a bit weird with flex-end

<img width="854" alt="image" src="https://github.com/marimo-team/marimo/assets/2753772/534c8a32-91e2-499f-b3bf-bf9ebc75f057">
<img width="810" alt="image" src="https://github.com/marimo-team/marimo/assets/2753772/1d322668-16c3-4e64-883c-0ca1428ab8c0">

akshayka commented:
> The groupby and aggregate can look a bit weird with flex-end
> 
> <img alt="image" width="854" src="https://user-images.githubusercontent.com/2753772/277847028-534c8a32-91e2-499f-b3bf-bf9ebc75f057.png"> <img alt="image" width="810" src="https://user-images.githubusercontent.com/2753772/277847044-1d322668-16c3-4e64-883c-0ca1428ab8c0.png">

Thanks for catching this! I've fixed this.

- I was trying to make the trash icon for deleting a form row appear inline with the controls. I think it mostly works now.
- To make it work I changed the `FormMessage` to be a tooltip in one place. Not tied to using a tooltip, but otherwise I was finding it hard to align everything. It's possible I regressed something, could use your help if I did. 
- Finally, I could use your help in repositioning the Drop NA for GroupBy, if possible. It'd be nice if it showed up in the bottom left or the top left, not in line with columns and aggregations as it does now.

<img width="527" alt="image" src="https://github.com/marimo-team/marimo/assets/1994308/70380c50-5771-4d02-9f56-847da17ebc08">

akshayka commented on frontend/src/components/ui/form.tsx:
```diff
@@ -176,6 +178,24 @@ const FormMessage = React.forwardRef<
+  }
+
+  return (
+    <TooltipProvider>
```
Not sure if this is the right place for the `Provider`. I was getting an error saying that `Tooltip` must be used in a provider. But come to think of it maybe that's because I was running in storybook
